drivers/serial/usart_gd32.c

```diff
@@ -38,6 +39,15 @@ static void usart_gd32_isr(const struct device *dev)
 }
 #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 
+static inline uint32_t usart_gd32_stat(uint32_t usart)
+{
+#ifdef CONFIG_SOC_SERIES_GD32F3X0
+	return USART_STAT(usart);
+#else
+	return USART_STAT0(usart);
+#endif /* CONFIG_SOC_SERIES_GD32F3X0 */
```

Comment:
I prefer use only `USART_STAT` in the whole code. This could be achieved with something like below at beginning of driver. This will eliminate many CONFIG_SOC_FOO again and keep code clean.

```
#ifndef USART_STAT
#define USART_STAT USART_STAT0 
#endif
```

Reply:
use IS_ENABLED instead? I think the redefine action will broken the connection between hal_gigadevice and usart_gd32.

Reply:
`IS_ENABLED` can't likely be used since it implies all branches can be built:

```c
if (IS_ENABLED(CONFIG_FOO)) {
     /* foo code */
} else {
     / * non-foo code */
}
```
^ foo and non-foo code must be buildable regardless of `CONFIG_FOO` value.

I think @nandojve approach makes sense.

Reply:
OK, use @nandojve approach.